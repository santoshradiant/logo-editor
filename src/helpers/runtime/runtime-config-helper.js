let runTimeConfig

// modify process env to new style
if (!runTimeConfig) {
  runTimeConfig = {}
}

function setRuntimeConfig (config) {
  runTimeConfig = config
}

function getRuntimeConfig () {
  return runTimeConfig
}

const appGateway = {
  development: 'https://api-gw.uat.builderservices.io',
  uat: 'https://api-gw.uat.builderservices.io',
  qa: 'https://api-gw.qa.builderservices.io',
  production: 'https://api-gw.builderservices.io'
}

const runtime = {
  _defaultApiName: 'logobuilder-api',
  version: 2,
  _brand: 'websitebuilder',
  _mfe: true,
  _fetchXsrfTokenHeaderName: 'X-CSRF-TOKEN',
  _env: process.env.NODE_ENV,
  _settings: { 'logomaker-brandname-readme-lock': false },
  _brandedUrls: {
    app_gateway: {
      api: appGateway[process.env.NODE_ENV]
    }
  }
}

const createRuntime = env => {
  return {
    ...runtime
  }
}

export { setRuntimeConfig, getRuntimeConfig, runtime, createRuntime }
