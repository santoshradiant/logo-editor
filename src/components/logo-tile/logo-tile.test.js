import 'jest-styled-components'
import 'core-js/stable'
import 'regenerator-runtime/runtime'
import React from 'react'
import { axe } from 'jest-axe'
import { render } from 'test-wrapper'

import Mocklogos from './__mock__/logos.json'
import LogoTileContainer from './logo-tile-container'
import LogoTile from './logo-tile'

describe('LogoTile', () => {
  test('Accessibility', async () => {
    const logo = Mocklogos[0]
    const selectLogo = jest.fn()
    const downloadLogo = jest.fn()
    const deleteLogo = jest.fn()
    const goEditLogo = jest.fn()
    const goDownloadLogo = jest.fn()
    const goPreviewLogo = jest.fn()
    const getNextLogo = jest.fn()
    const animateLogoBeforeNavigate = jest.fn()

    const { container } = await render(
      <div style={{ height: 500 }}>
        <LogoTileContainer key={logo.id}>
          <LogoTile
            logo={logo}
            makeLogoSelected={selectLogo}
            downloadLogo={downloadLogo}
            deleteLogo={deleteLogo}
            goEditLogo={goEditLogo}
            goDownloadLogo={goDownloadLogo}
            goPreviewLogo={goPreviewLogo}
            getNextLogo={getNextLogo}
            // set animation off for now
            animateLogoBeforeNavigate={animateLogoBeforeNavigate}
          />
        </LogoTileContainer>
      </div>
    )

    // Compare snapshot
    expect(container).toMatchSnapshot()

    // Check accessibility
    const result = await axe(container)
    expect(result).toHaveNoViolations()
  })
})
