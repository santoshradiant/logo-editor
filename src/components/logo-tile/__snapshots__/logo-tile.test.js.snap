// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoTile Accessibility 1`] = `
<div>
  <div
    class="app-logo"
  >
    <div
      class="app-logo-NotificationBar-fullScreenVariant-1"
    />
    <div
      style="height: 500px;"
    >
      <div
        class="app-logo-MuiGrid-root app-logo-MuiGrid-item app-logo-MuiGrid-grid-xs-12 app-logo-MuiGrid-grid-sm-12 app-logo-MuiGrid-grid-md-4 app-logo-MuiGrid-grid-lg-4 app-logo-MuiGrid-grid-xl-3"
      >
        <div
          class="logo-tileviews__LogoTileContainer-sc-11u1tqa-4 jPBmYk"
          placeholder="0"
        >
          <div
            class="sc-fznxsB dkwiiY     "
            data-element-id="logomaker.logoTile.Card"
            data-element-label="logomaker.logoTile.Card"
            data-element-location="left_rail"
            data-element-type="button"
          >
            <div
              class="app-logo-MuiPaper-root app-logo-MuiCard-root logo-tileviews__StyledCard-sc-11u1tqa-0 bLpfqz app-logo-CustomCard-default-4 app-logo-MuiPaper-elevation1 app-logo-MuiPaper-rounded"
              placeholder="0"
            >
              <div
                class="logo-tileviews__ImageContainer-sc-11u1tqa-5 fqDcGl"
              >
                <img
                  alt="logo-preview"
                  class="logo-placeholder-wrapper__StyledImage-ytorvw-0 gWEksp my-logo-preview"
                  src="https://lmuserassetsdev.blob.core.windows.net/account-105-public/5344939a28b14b6ea6ed7d933ea3dc35.png"
                />
              </div>
            </div>
          </div>
          <div
            class="logo-tileviews__ActionCardHoverBox-sc-11u1tqa-2 kWXwZb"
          >
            <div
              class="app-logo-MuiPaper-root app-logo-MuiCard-root logo-tileviews__ActionCard-sc-11u1tqa-3 bqCLpr app-logo-CustomCard-default-4 app-logo-MuiPaper-elevation1 app-logo-MuiPaper-rounded"
              data-test-id="logo-action-card"
            >
              <div
                class="app-logo-MuiGrid-root app-logo-MuiGrid-container app-logo-MuiGrid-align-items-xs-center app-logo-MuiGrid-justify-xs-space-around"
              >
                <div
                  class="app-logo-MuiGrid-root logo-tileviews__CenterGridItem-sc-11u1tqa-1 jLuffC app-logo-MuiGrid-item app-logo-MuiGrid-grid-xs-true"
                >
                  <button
                    alt="edit"
                    aria-label="edit"
                    class="app-logo-MuiButtonBase-root app-logo-MuiIconButton-root"
                    data-element-id="logomaker-my-logos-edit-button"
                    data-element-label="my-logos-preview"
                    data-element-location="header"
                    data-element-type="button"
                    tabindex="0"
                    title="edit"
                    type="button"
                  >
                    <span
                      class="app-logo-MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="app-logo-MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
                <div
                  class="app-logo-MuiGrid-root logo-tileviews__CenterGridItem-sc-11u1tqa-1 jLuffC app-logo-MuiGrid-item app-logo-MuiGrid-grid-xs-true"
                >
                  <button
                    alt="preview"
                    aria-label="preview"
                    class="app-logo-MuiButtonBase-root app-logo-MuiIconButton-root"
                    data-element-id="logomaker-my-logos-preview-button"
                    data-element-label="my-logos-preview"
                    data-element-location="header"
                    data-element-type="button"
                    tabindex="0"
                    title="preview"
                    type="button"
                  >
                    <span
                      class="app-logo-MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="app-logo-MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                        />
                        <path
                          d="M12 10h-2v2H9v-2H7V9h2V7h1v2h2v1z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
                <div
                  class="app-logo-MuiGrid-root logo-tileviews__CenterGridItem-sc-11u1tqa-1 jLuffC app-logo-MuiGrid-item app-logo-MuiGrid-grid-xs-true"
                >
                  <button
                    alt="download"
                    aria-label="download"
                    class="app-logo-MuiButtonBase-root app-logo-MuiIconButton-root"
                    data-element-id="logomaker-my-logos-download-button"
                    data-element-label="my-logos-preview"
                    data-element-location="header"
                    data-element-type="button"
                    tabindex="0"
                    title="download"
                    type="button"
                  >
                    <span
                      class="app-logo-MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="app-logo-MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM17 13l-5 5-5-5h3V9h4v4h3z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
                <div
                  class="app-logo-MuiGrid-root logo-tileviews__CenterGridItem-sc-11u1tqa-1 jLuffC app-logo-MuiGrid-item app-logo-MuiGrid-grid-xs-true"
                >
                  <button
                    alt="duplicate"
                    aria-label="duplicate"
                    class="app-logo-MuiButtonBase-root app-logo-MuiIconButton-root"
                    data-element-id="logomaker-my-logos-duplicate-button"
                    data-element-label="my-logos-preview"
                    data-element-location="header"
                    data-element-type="button"
                    tabindex="0"
                    title="duplicate"
                    type="button"
                  >
                    <span
                      class="app-logo-MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="app-logo-MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm16-4H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-1 9h-4v4h-2v-4H9V9h4V5h2v4h4v2z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
                <div
                  class="app-logo-MuiGrid-root logo-tileviews__CenterGridItem-sc-11u1tqa-1 jLuffC app-logo-MuiGrid-item app-logo-MuiGrid-grid-xs-true"
                >
                  <button
                    alt="delete"
                    aria-label="delete"
                    class="app-logo-MuiButtonBase-root app-logo-MuiIconButton-root"
                    data-element-id="logomaker-my-logos-delete-button"
                    data-element-label="my-logos-preview"
                    data-element-location="header"
                    data-element-type="button"
                    tabindex="0"
                    title="delete"
                    type="button"
                  >
                    <span
                      class="app-logo-MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="app-logo-MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
